build.zig

```diff
@@ -127,36 +129,54 @@ pub fn build(b: *std.build.Builder) !void {
-        b.pathFromRoot("README.md"),
-        "--generate-config-path",
-        b.pathFromRoot("src/Config.zig"),
-        "--generate-schema-path",
-        b.pathFromRoot("schema.json"),
-    });
-    if (b.args) |args| gen_cmd.addArgs(args);
-
-    const gen_step = b.step("gen", "Regenerate config files");
-    gen_step.dependOn(&gen_cmd.step);
+    const gen_config_cmd = gen_config_cmd: {
+        const cmd = b.addRunArtifact(gen_exe);
+        cmd.addArgs(&.{
+            "--readme-path",
+            b.pathFromRoot("README.md"),
+            "--generate-config-path",
+            b.pathFromRoot("src/Config.zig"),
+            "--generate-schema-path",
+            b.pathFromRoot("schema.json"),
+        });
+        if (b.args) |args| cmd.addArgs(args);
+        break :gen_config_cmd cmd;
+    };
 
-    const gen_version_data_cmd = b.addRunArtifact(gen_exe);
-    gen_version_data_cmd.addArgs(&.{ "--generate-version-data", data_version });
-    if (data_version_path) |path| {
-        gen_version_data_cmd.addArg("--langref_path");
-        gen_version_data_cmd.addFileArg(.{ .path = path });
-    }
-    const version_data_file_name = if (data_version_path != null)
-        b.fmt("version_data_{s}.zig", .{data_version})
-    else blk: {
-        // invalidate version data periodically from cache because the website content may change
-        // setting `has_side_effects` would also be possible but that would always force a re-run
-        const timestamp = @divFloor(std.time.timestamp(), std.time.s_per_day);
-        break :blk b.fmt("version_data_{s}_{d}.zig", .{ data_version, timestamp });
+    const gen_config_step = b.step("gen-config", "Regenerate config files and README");
+    gen_config_step.dependOn(&gen_config_cmd.step);
+
+    const version_data_path: std.Build.LazyPath = version_data_path: {
+        const file_name = if (override_version_data_file_path) |override_path|
+            override_path
+        else if (override_langref_file_path != null)
+            b.fmt("version_data_{s}.zig", .{compiler_version})
+        else blk: {
+            // invalidate version data periodically from cache because the website content may change
+            // setting `has_side_effects` would also be possible but that would always force a re-run
+            const timestamp = @divFloor(std.time.timestamp(), std.time.s_per_day);
+            break :blk b.fmt("version_data_{s}_{d}.zig", .{ compiler_version, timestamp });
+        };
+        break :version_data_path .{ .path = file_name };
+    };
+    
+    const gen_version_data_cmd = gen_version_data_cmd: {
+        const cmd = b.addRunArtifact(gen_exe);
+        cmd.addArgs(&.{ "--generate-for-version", compiler_version });
+        if (override_langref_file_path) |override_path| {
+            cmd.addArg("--langref_path");
+            cmd.addFileArg(.{ .path = override_path });
+        }
+        cmd.addArg("--generate-version-data-path");
+        cmd.addFileArg(version_data_path);
```

Comment:
> What you want to change is that if override_version_data_file_path is set, then there is no dependency on running gen_version_data_cmd because you already have the version data file.

This behaviour (lazy evualted and no dependency on running) is also observed in my version (see lines below and above), but OK, I reverted it anyway.

Reply:
You achieved this by manually declaring the dependency [here](https://github.com/zigtools/zls/blob/1ff45b72b6dea1bf14e66591f772d697dff70a32/build.zig#L178) which shoudn't be necessary because `std.Build.LazyPath` combined with `addOutputFileArg` can be used to declare that a path depends on a step.